src/plugins/saved_objects/public/finder/saved_object_finder.tsx

```diff
@@ -172,13 +201,24 @@ class SavedObjectFinderUi extends React.Component<
             getDataSource
           );
           return result;
+        } else if (obj.type === 'search') {
+          const sourceObject = JSON.parse(
+            obj.attributes?.kibanaSavedObjectMeta?.searchSourceJSON ?? null
+          );
+          const languageId = sourceObject?.query?.language;
+          if (this.isSavedSearchLanguageSupported(languageId, currentAppId, this.languageService)) {
```

Comment:
i wonder if it makes it a little bit more scalable/safer to just have this as a pass a function

like from the dashboards it could be

```typescript
 SavedObjectFinder: getSavedObjectFinder(
          savedObjects,
          uiSettings,
          searchFilterFn: (obj) => {
              const sourceObject = JSON.parse(
              obj.attributes?.kibanaSavedObjectMeta?.searchSourceJSON ?? null);
              const languageId = sourceObject?.query?.language;
              services.data.languageService?.getLanguage(languageId)?.supportedAppNames?.includes(currentAppId) ?? true;
          }
      )
```

Reply:
i think i only worry ensuring we dont end up adding more required plugins since i know a good amount of community members hit the APIs directly and even though this is the public side the saved objects plugin has been historically didn't need to care about the application which made it reliably abstracted and enforced a pattern for developers.

do we forsee in issues with introducing these concepts of language service to the saved object finder?

Reply:
The overall change seems safe and the default behavior without the additional parameters remains unchanged for SavedObjectFinder. The saved_object_finder.test.tsx tests all use the default version of SavedObjectFinder without the optional parameters and the behavior remains unchanged.
 
In the meanwhile I have created the issue https://github.com/opensearch-project/OpenSearch-Dashboards/issues/8122 , which we can use to create a more scalable version of the above when required.